src/pubd/repository.rs

```diff
@@ -1363,17 +1377,81 @@ impl PublisherStats {
         self.size
     }
 
+    /// Returns the most recent "this_update" time
+    /// from all manifest(s) published by this publisher,
+    /// if any.. i.e. there may be 0, 1 or many manifests
     pub fn last_update(&self) -> Option<Time> {
-        self.last_update
+        let mut last_update = None;
+        for mft in self.manifests() {
+            if let Some(seen_update) = last_update {
```

Comment:
Introducing a new `seen_update` is confusing, why not just shape the existing `last_update`? Ah I see, because of the assignment... still I found it confusing.

Reply:
renamed things more explicitly - hopefully improving readability